n3fit/src/evolven3fit_new/utils.py

```diff
@@ -0,0 +1,179 @@
+import shutil
+import pathlib
+from scipy.interpolate import interp1d
+import numpy as np
+from reportengine.compat import yaml
+from validphys.pdfbases import PIDS_DICT
+
+
+class LhapdfLike:
+    """
+    Class which emulates lhapdf but only for an initial condition PDF (i.e. with only one q2 value)
+    """
+
+    def __init__(self, pdf_grid, q20, x_grid):
+        self.pdf_grid = pdf_grid
+        self.q20 = q20
+        self.x_grid = x_grid
+        self.funcs = [
+            interp1d(self.x_grid, self.pdf_grid[pid], kind="cubic")
+            for pid in range(len(PIDS_DICT))
+        ]
+
+    def xfxQ2(self, pid, x, q2):
+        """Return the value of the PDF for the requested pid and x value. If the requested q2
+        value is different from the (only) value available, it raises an error.
+
+        Parameters
+        ----------
+
+            pid: int
+                pid index of particle
+            x: float
+                x-value
+            q2: float
+                Q square value
+
+        Returns
+        -------
+            : float
+            x * PDF value
+        """
+        if not np.isclose(q2, self.q20, rtol=1e-6):
```

Comment:
Indeed. This was requested by @scarlehoff. I agree with you that, since the two values are both coming from the same source (the tcard) they should always be the exact same. But since this was requested by @scarlehoff maybe he has a reason to prefer this.

Reply:
The reason is that either you expected them to be the same and only the same (and then don't test at all) or you expect some changes for whatever reason and then a tolerance should be given. But I would rather not have this test at all.

Reply:
For me we can even drop the test: it is there only to be sure we are not doing something silly.

Reply:
What I don't understand is why don't you use self.q20 if you already have it at that point.

Reply:
Yes exactly, it is called by eko. So I really need a function with a q2 entry (an lhapdf-like function).

Reply:
I see. Then either remove the test altogether (if someone is calling that, they are getting the PDF at q2) or the tolerance is needed. We don't want `1.65000000000000 and 1.650000000000001` being different because between `numpy` and `numba` and whatsnot there was a missing bit somewhere.